server/fleet/sessions.go

```diff
@@ -49,3 +71,104 @@ type Session struct {
 func (s Session) AuthzType() string {
 	return "session"
 }
+
+// SSORolesInfo holds the configuration parsed from SAML custom attributes.
+//
+// `Global` and `Teams` are never both set (by design, users must be either global
+// or member of teams).
+type SSORolesInfo struct {
+	// Global holds the role for the Global domain.
+	Global *string
+	// Teams holds the roles for teams.
+	Teams []TeamRole
+}
+
+// TeamRole holds a user's role on a Team.
+type TeamRole struct {
+	// ID is the unique identifier of the team.
+	ID uint
+	// Role is the role of the user in the team.
+	Role string
+}
+
+func (s SSORolesInfo) verify() error {
+	if s.Global != nil && len(s.Teams) > 0 {
+		return errors.New("cannot set both global and team roles")
+	}
+	// Check for duplicate entries for the same team.
+	// This is just in case some IdP allows duplicating attributes.
+	teamMap := make(map[uint]struct{})
+	for _, teamRole := range s.Teams {
+		if _, ok := teamMap[teamRole.ID]; ok {
+			return fmt.Errorf("duplicate team entry: %d", teamRole.ID)
+		}
+		teamMap[teamRole.ID] = struct{}{}
+	}
+	return nil
+}
+
+func (s SSORolesInfo) empty() bool {
```

Comment:
Consider `isEmpty`.

`empty` could be understood as a verb. (Empty this struct = make it empty)